### PR TITLE
ci: upsize many slow-running circleci builds

### PR DESCRIPTION
**Pull request by rboyer**

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -241,7 +244,7 @@ jobs:
   go-test-arm64:
     machine:
       image: ubuntu-2004:202101-01
-    resource_class: arm.medium
+    resource_class: arm.large
```
Comment (rboyer) [inline .circleci/config.yml:247]: `10m 25s -> 7m`

File: .circleci/config.yml [hunk 2]
```diff
@@ -272,6 +275,7 @@ jobs:
   go-test:
     docker:
       - image: *GOLANG_IMAGE
+    resource_class: large
```
Comment (rboyer) [inline .circleci/config.yml:278]: `6m 30s -> 5m 30s`

File: .circleci/config.yml [hunk 3]
```diff
@@ -328,6 +333,7 @@ jobs:
   go-test-32bit:
     docker:
       - image: *GOLANG_IMAGE
+    resource_class: large
```
Comment (rboyer) [inline .circleci/config.yml:336]: `7m 16s -> 5m 29s`

File: .circleci/config.yml [hunk 4]
```diff
@@ -367,6 +373,7 @@ jobs:
         type: string
     docker:
       - image: "docker.mirror.hashicorp.services/cimg/go:<<parameters.go-version>>"
+    resource_class: large
```
Comment (rboyer) [inline .circleci/config.yml:376]:
```suggestion
```

File: .circleci/config.yml [hunk 5]
```diff
@@ -470,6 +479,7 @@ jobs:
   dev-build:
     docker:
       - image: *GOLANG_IMAGE
+    resource_class: large
```
Comment (rboyer) [inline .circleci/config.yml:482]: `2m 37s -> 1m 41s`

File: .circleci/config.yml [hunk 6]
```diff
@@ -582,6 +593,7 @@ jobs:
   nomad-integration-main:
     docker:
       - image: docker.mirror.hashicorp.services/circleci/golang:1.17 # TODO: replace with cimg/go (requires steps update)
+    resource_class: large
```
Comment (rboyer) [inline .circleci/config.yml:596]:
```suggestion
```

File: .circleci/config.yml [hunk 7]
```diff
@@ -843,7 +855,7 @@ jobs:
     machine:
       image: ubuntu-2004:202101-01
     parallelism: 4
-    resource_class: medium
+    resource_class: large
```
Comment (rboyer) [inline .circleci/config.yml:858]:
```suggestion
    resource_class: medium
```

**Comment by hc-github-team-consul-core** [comment]
:cherries: If backport labels were added before merging, cherry-picking will start automatically.

To retroactively trigger a backport after merging, add backport labels and re-run https://circleci.com/gh/hashicorp/consul/633682.

**Comment by hc-github-team-consul-core** [comment]
:cherries::white_check_mark: Cherry pick of commit c24df504762c23ec83def40d96eb83b04e0edcd5 onto `release/1.12.x` succeeded!

**Comment by hc-github-team-consul-core** [comment]
:cherries::x: Cherry pick of commit c24df504762c23ec83def40d96eb83b04e0edcd5 onto `release/1.11.x` failed! [Build Log](https://circleci.com/gh/hashicorp/consul/633682)

**Comment by hc-github-team-consul-core** [comment]
:cherries::x: Cherry pick of commit c24df504762c23ec83def40d96eb83b04e0edcd5 onto `release/1.10.x` failed! [Build Log](https://circleci.com/gh/hashicorp/consul/633682)

**Pull request merged**